### PR TITLE
feat(LERN-98): implemented createStudent endpoint and integration tests

### PR DESCRIPTION
**Pull request by A1eeee**

### CODE REVIEW

**Comment by lost-pixel-signals** [comment]
| Shot Type       | Count                | Status   |
| -------------   | -------------------- | -------- |
| Added           | 0       | 🟡       |
| Deleted         | 0     | ❌       |
| Regression      | 1  | ❗       |
    


*❗ custom/admin-landing-page*: 


| Before | After |
| --- | --- |
|  |  |

**Comment by lost-pixel-signals** [comment]
| Shot Type       | Count                | Status   |
| -------------   | -------------------- | -------- |
| Added           | 0       | 🟡       |
| Deleted         | 0     | ❌       |
| Regression      | 1  | ❗       |
    


*❗ custom/admin-landing-page*: 


| Before | After |
| --- | --- |
|  |  |

File: backend/src/main/kotlin/ch/learntrack/backend/user/UserResource.kt [hunk 1]
```diff
@@ -35,7 +36,7 @@ public class UserResource(
 
     @PostMapping("/register")
     public fun register(@RequestBody createUserDto: CreateUserDto): LoginResponseDto {
-        val user = userService.createUser(createUserDto)
+        val user = userService.createUser(createUserDto, UserRole.STUDENT, null, null)
```
Comment (deepspace1000) [inline backend/src/main/kotlin/ch/learntrack/backend/user/UserResource.kt:39]: UserRole.STUDENT?? that looks not right shouldn't that be the teacher register endpoint for teachers?

Comment (deepspace1000) [reply backend/src/main/kotlin/ch/learntrack/backend/user/UserResource.kt:39]: have a look at the teachers registrations tests. are there no test which check the UserRole of the teacher after its created? if not add one.

**Review by deepspace1000 — COMMENTED** [review]
Looks good to me mostly. just this one part i think you mixed up.

**Comment by lost-pixel-signals** [comment]
| Shot Type       | Count                | Status   |
| -------------   | -------------------- | -------- |
| Added           | 0       | 🟡       |
| Deleted         | 0     | ❌       |
| Regression      | 0  | ❗       |

File: backend/src/main/kotlin/ch/learntrack/backend/admin/student/AdminStudentService.kt [hunk 2]
```diff
@@ -16,4 +17,17 @@ public class AdminStudentService(
                 grade = gradeService.getGradesForUserByUserId(user.id).first(),
             )
         }
+
+    public fun createStudent(createStudentDto: CreateStudentDto) {
+        val createUserDto = CreateUserDto(
```
Comment (deepspace1000) [inline backend/src/main/kotlin/ch/learntrack/backend/admin/student/AdminStudentService.kt:22]: I don't really like it but in lack of a besser idea i approve 😅

**Review by deepspace1000 — APPROVED** [review]
If half of this 400 lines of test work i think this code should do what it should.

**Comment by lost-pixel-signals** [comment]
| Shot Type       | Count                | Status   |
| -------------   | -------------------- | -------- |
| Added           | 0       | 🟡       |
| Deleted         | 0     | ❌       |
| Regression      | 0  | ❗       |

**Review by deepspace1000 — APPROVED** [review]
LGTM

**Comment by lost-pixel-signals** [comment]
| Shot Type       | Count                | Status   |
| -------------   | -------------------- | -------- |
| Added           | 0       | 🟡       |
| Deleted         | 0     | ❌       |
| Regression      | 1  | ❗       |
    


*❗ custom/login-page*: 


| Before | After |
| --- | --- |
|  |  |

**Pull request merged**